api/pkg.go

```diff
@@ -0,0 +1,120 @@
+package api
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/konveyor/controller/pkg/logging"
+	"github.com/konveyor/tackle-hub/settings"
+	"gorm.io/gorm"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+var (
+	Settings = &settings.Settings
+	log      = logging.WithName("api")
+)
+
+//
+// Routes
+const (
+	InventoryRoot = "/application-inventory"
+	ControlsRoot  = "/controls"
+)
+
+//
+// Params
+const (
+	ID       = "id"
+	Name     = "name"
+	Wildcard = "Wildcard"
+)
+
+//
+// Pagination Defaults
+const (
+	Limit  = 20
+	Offset = 0
+	Sort   = "id asc"
+)
+
+//
+// All builds all handlers.
+func All() []Handler {
+	return []Handler{
+		&ApplicationHandler{},
+		&BucketHandler{},
+		&BusinessServiceHandler{},
+		&DependencyHandler{},
+		&ImportHandler{},
+		&JobFunctionHandler{},
+		&RepositoryHandler{},
+		&IdentityHandler{},
+		&ReviewHandler{},
+		&StakeholderHandler{},
+		&StakeholderGroupHandler{},
+		&TagHandler{},
+		&TagTypeHandler{},
+		&TaskHandler{},
+		&AddonHandler{},
+	}
+}
+
+//
+// Handler.
+type Handler interface {
+	With(*gorm.DB, client.Client)
+	AddRoutes(e *gin.Engine)
+}
+
+//
+// Pagination provides pagination and sorting.
+type Pagination struct {
```

Comment:
Not sure this belongs in _pkg_ since it's not intended to be used outside the package.  Probably better left in base or in a new pagination.go.